pinot-spi/src/main/java/org/apache/pinot/spi/config/table/ingestion/FilterConfig.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.config.table.ingestion;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import javax.annotation.Nullable;
+import org.apache.pinot.spi.config.BaseJsonConfig;
+
+
+/**
+ * Configs related to filtering records during ingestion
+ */
+public class FilterConfig extends BaseJsonConfig {
+
+  @JsonPropertyDescription("Filter function string. Filter out records during ingestion, if this evaluates to true")
+  private final String _filterFunction;
```

Comment:
Shouldn't this be a list of filter functions applied in sequence specified? How do we indicate filter like:
1. Sample x% of the records
2. filter out those with age > 40

Or, 
1. Filter out those with age > 40
2. Sample x% of these records.

Reply:
If there's more than 1 clause, they can be put into the same function using && ||. Added an example in the description and also in the docs: https://docs.pinot.apache.org/developers/advanced/ingestion-level-transformations

Reply:
Can the sample percent be a function?
`"filterFunction": "samplePercent() < 40"`
Basically an inbuilt function with no arguments.

Reply:
If our transform function evaluator supported this expression, we could support this. But it only allows function calls. To support this, I believe we would need something like
`filterFunction: "samplePercent(40)"`

Reply:
we can add it later, just wanted to know what the interface would look like if we added a function. So, in this case, the samplePercent(n) should return a boolean?

Reply:
that's right. final return value should be a boolean.
could also be `isLessThan(samplePercent(), 40)` which maps closer to your example. We probably would be adding a lot of such basic inbuilt functions soon anyway.